pkg/agent/kubermatic/v1/agent.go

```diff
@@ -55,27 +58,34 @@ func NewAgent(client client.Client, info serverVersionInfo, dataStore datastore.
 	}
 }
 
-// +kubebuilder:rbac:groups="kubermatic.k8c.io",resources=seeds;clusters;users;projects;usersshkeys;kubermaticconfigurations,verbs=list
+// +kubebuilder:rbac:groups="kubermatic.k8c.io",resources=seeds;clusters;users;projects;usersshkeys,verbs=list
+// +kubebuilder:rbac:groups="kubermatic.k8c.io",resources=kubermaticconfigurations,verbs=get
 // +kubebuilder:rbac:groups="",resources=secrets,verbs=get
 
 func (a kubermaticAgent) Collect(ctx context.Context) error {
-	serverVersion, err := a.ServerVersion()
-	if err != nil {
-		return err
-	}
 	record := Record{
 		KindVersion: agent.KindVersion{
 			Kind:    "kubermatic",
 			Version: "v1",
 		},
-		Time:              time.Now().UTC(),
-		KubernetesVersion: serverVersion.String(),
+		Time: time.Now().UTC(),
 	}
 
-	defaultExposeStrategy, err := a.getDefaultExposeStrategy(ctx)
-	if err != nil {
+	// Get Kubermatic Configuration
+	config := &kubermaticv1.KubermaticConfiguration{}
+	if err := a.Get(ctx, types.NamespacedName{Name: "kubermatic", Namespace: resources.KubermaticNamespace}, config); err != nil {
```

Comment:
The configuration is not necessarily named "kubermatic". KKP lists all available KCs, exits with an error if 0 or more than 1 are found, and otherwise takes the one that it found, no matter how it's named.

You can use a `provider.DynamicKubermaticConfigurationGetterFactory` and call the getter afterwards to be compatible with KKP.

Reply:
Already pushed the code hours ago 🤔

Reply:
Oh sorry ^^